### PR TITLE
Fix/190

### PR DESCRIPTION
**Pull request by xross**
Addresses issue #190 

Tests can now be run over multiple processes. Essentially we now build to different bin location based on params. Previously we cleaned and rebuilt to the same bin location with different defines.

There is definitely room for tidy up here, but the scheme works essentially. We could also move the build step to after the collection step now if we liked (so build all tests, then run all tests, rather than build just in time)

Running smoke tests single vs 4 processes (pytest --smoke -n=4) gives about 50% speed up. 

Note, using -n=auto is recommenced

### CODE REVIEW

File: tests/shared_src/test_main.xc [hunk 1]
```diff
@@ -43,6 +43,7 @@ int main()
                 XUD_Kill(ep_out_0);
             }
             exit(0);
+		}
```
Comment (DaveAtkinsonXMOS) [inline tests/shared_src/test_main.xc:46]: Um, is this indented correctly?

**Review by DaveAtkinsonXMOS — APPROVED** [review]
One questionable bit of indentation, not really a show stopper...
Tick...

File: tests/shared_src/test_main.xc (hunk 1 above)
Comment (xross) [reply tests/shared_src/test_main.xc]: Looks like a tab crept in.. Fixed.

**Review by DaveAtkinsonXMOS — COMMENTED** [review]
Nice...

(submitting this comment to close review on merged PR).

**Pull request merged**